iotdb-1.3/src/main/java/cn/edu/tsinghua/iot/benchmark/iotdb130/RegisterSchema.java

```diff
@@ -0,0 +1,329 @@
+package cn.edu.tsinghua.iot.benchmark.iotdb130;
+
+import cn.edu.tsinghua.iot.benchmark.conf.Config;
+import cn.edu.tsinghua.iot.benchmark.conf.ConfigDescriptor;
+import cn.edu.tsinghua.iot.benchmark.entity.Sensor;
+import cn.edu.tsinghua.iot.benchmark.entity.enums.SensorType;
+import cn.edu.tsinghua.iot.benchmark.schema.schemaImpl.DeviceSchema;
+import cn.edu.tsinghua.iot.benchmark.tsdb.DBConfig;
+import cn.edu.tsinghua.iot.benchmark.tsdb.IDatabase;
+import cn.edu.tsinghua.iot.benchmark.tsdb.TsdbException;
+import cn.edu.tsinghua.iot.benchmark.utils.TimeUtils;
+import org.apache.iotdb.isession.template.Template;
+import org.apache.iotdb.isession.util.Version;
+import org.apache.iotdb.rpc.IoTDBConnectionException;
+import org.apache.iotdb.rpc.StatementExecutionException;
+import org.apache.iotdb.session.Session;
+import org.apache.iotdb.session.template.MeasurementNode;
+import org.apache.iotdb.tsfile.file.metadata.enums.CompressionType;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSEncoding;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+
+public abstract class RegisterSchema implements IDatabase {
+
+    private static final String ALREADY_KEYWORD = "already";
+    private static final AtomicBoolean templateInit = new AtomicBoolean(false);
+    protected final Logger LOGGER;
+    protected SingleNodeJDBCConnection ioTDBConnection;
```

Comment:
I think we need to consider whether extract the logic of registering schema out of IoTDB. Maybe we need to implement RestAPi extends IoTDB
 and override related method?

Reply:
Maybe later we will implement register schema in different connection type~